src/services/account-states/account-state-engine.ts

```diff
@@ -1,118 +1,189 @@
+    if (!allowedTransition)
+      throw buildError(
+        MetricNames.NO_TRANSITIONS_FOUND_IN_CONFIG,
+        `There are no allowed transitions from state ${nodeKye} in current configurations`,
+      );
+    return allowedTransition;
+  }
 
-      const newStateDetails = AccountStateEngine.interventionConfigurations[proposedTransition];
-      if (!newStateDetails) {
-        logAndPublishMetric(MetricNames.INTERVENTION_EVENT_NOT_FOUND_IN_CURRENT_CONFIG);
-        throw new StateTransitionError(`Intervention ${proposedTransition} cannot be found in current config.`);
-      }
-      interventionName = newStateDetails.interventionName;
-      stateChange = newStateDetails.state;
+  /**
+   * Helper method to get the transition code from a given transition name
+   * if the passed transition is in the given list of allowed transitions it returns the code
+   * it throws an error otherwise
+   *
+   * @param allowedTransition - list of allowed transition codes
+   * @param transition - EventsEnum representation of the proposed transition
+   */
+  private getTransition(allowedTransition: number[], transition: EventsEnum) {
+    for (const element of allowedTransition) {
+      if (AccountStateEngine.configuration.edges[element]?.name === transition.toString()) return element;
     }
+    throw buildError(
+      MetricNames.STATE_TRANSITION_NOT_ALLOWED_OR_IGNORED,
+      `${transition} is not allowed from current state`,
+    );
+  }
 
-    const newState = Object.assign({ ...currentAccountStateDetails }, stateChange);
-    return buildPartialUpdateAccountStateCommand(newState, proposedTransition, interventionName);
+  /**
+   * Helper method to return the new account state given a transition
+   * @param edge - code mapping to a specific transition
+   */
+  private getNewStateObject(edge: number) {
+    const newStateName = AccountStateEngine.configuration.edges[edge]!.to;
+    const newStateObject = AccountStateEngine.configuration.nodes[newStateName];
+    if (!newStateObject)
+      throw buildError(
+        MetricNames.STATE_NOT_FOUND_IN_CURRENT_CONFIG,
+        `state ${newStateName} not found in current config.`,
+      );
+    return newStateObject;
   }
 }
+
+/**
+ * Helper method to build a StateTransitionError and log relevant information when an error is to br thrown
+ * @param metricName - name of the metric to log
+ * @param errorMessage - error message to be logged
+ */
+function buildError(metricName: MetricNames, errorMessage: string) {
+  logAndPublishMetric(metricName);
+  logger.error({ message: errorMessage });
+  return new StateTransitionError(errorMessage);
+}
+
+/**
+ * Helper method to compare two StateDetails objects.
+ * It returns true, IFF all fields are the same, false otherwise
+ * @param aState - first account state
+ * @param anotherState - second account state
+ */
+function compareAccountStates(aState: StateDetails, anotherState: StateDetails) {
```

Comment:
I wonder if changing the function name to something like `isAccountStatesTheSame` (or `areAccountStatesTheSame` would be better english) would make it clearer what the return value means